swatch-contracts/src/main/java/com/redhat/swatch/contract/service/AwsPartnerEntitlementsProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.contract.service;
+
+import com.redhat.swatch.clients.rh.partner.gateway.api.model.QueryPartnerEntitlementV1;
+import com.redhat.swatch.contract.openapi.model.PartnerEntitlementContract;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.Objects;
+
+@ApplicationScoped
+public class AwsPartnerEntitlementsProvider extends BasePartnerEntitlementsProvider {
+
+  @Override
+  public boolean isValid(PartnerEntitlementContract contract) {
+    return Objects.nonNull(contract.getRedHatSubscriptionNumber())
+        && Objects.nonNull(contract.getCloudIdentifiers())
+        && Objects.nonNull(contract.getCloudIdentifiers().getAwsCustomerId())
+        && Objects.nonNull(contract.getCloudIdentifiers().getAwsCustomerAccountId())
+        && Objects.nonNull(contract.getCloudIdentifiers().getProductCode());
+  }
```

Comment:
Could we use bean validation here instead?  I bring it up because @kahowell just retired some very similar code in #3050 [here](https://github.com/RedHatInsights/rhsm-subscriptions/pull/3050/files#diff-d0a3f94bef59f416c81d8ea1f761cdfcd6e5ae3e5888e580e5b398772ce8b5f1L499-L509).  I realize it's a bit tricky since the `PartnerEntitlementContract` is validated in different ways depending on context, but maybe it's worth a little research into whether there's a solution with bean validation here.

Just taking a quick look, it seems like there might be away to apply different validation rules at [runtime](https://docs.jboss.org/hibernate/stable/validator/reference/en-US/html_single/#_groupsequenceprovider).  Chapter 5 in that document seems broadly relevant.  We could have validation groups for `CloudIdentifiers`: one for Azure, one for AWS, and one default (which would be a `NonNull` on `getProductCode`) and then apply the correct groups based on a `GroupSequenceProvider` implementation.

Reply:
https://quarkus.io/guides/validation#validation-groups-for-rest-endpoint-or-service-method-validation ?

Reply:
This is not a validation method, but a way to identify if it's for azure or aws. The `isValid` method name is misleading, will rename it.